### PR TITLE
initial support for txn

### LINKED ISSUE #3283: v3 txn API support
We need to add a draft version of txn support.

### PR DESCRIPTION
**Pull request by xiang90**
Fix #3283

example

```bash
# execute put
./etcdctlv3 put foo bar
foo bar

# execute txn
./etcdctlv3 txn
entry comparison[key target expected_result compare_value] (end with empty line):
foo value equal bar
entry comparison[key target expected_result compare_value] (end with empty line):

entry success request[method key value(end_range)] (end with empty line):
put foo bar2
entry success request[method key value(end_range)] (end with empty line):
put foo bar3
entry success request[method key value(end_range)] (end with empty line):

entry failure request[method key value(end_range)] (end with empty line):
put foo bar4
entry failure request[method key value(end_range)] (end with empty line):

executed success request list

# execute range
./etcdctlv3 range foo
foo bar3
```

### CODE REVIEW

File: etcdctlv3/command/txn_command.go [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+	if len(parts) < 2 {
+		return nil, fmt.Errorf("invalid txn compare request: %s", line)
+	}
+
+	ru := &pb.RequestUnion{}
+	key := []byte(parts[1])
+	switch parts[0] {
+	case "r":
+		fallthrough
+	case "range":
```
Comment (yichengq) [inline etcdctlv3/command/txn_command.go]: `case "r", "range":`?

File: etcdserver/v3demo_server.go [hunk 2]
```diff
@@ -27,36 +30,135 @@ type V3DemoServer interface {
...
+		}
+		if len(resps) != 0 {
+			index += 1
+		}
+
+		txnResp := &pb.TxnResponse{}
+		txnResp.Header = &pb.ResponseHeader{}
+		txnResp.Header.Index = index
+		txnResp.Responses = resps
+		txnResp.Succeeded = ok
```
Comment (yichengq) [inline etcdserver/v3demo_server.go]: put it in a struct?

**Comment by yichengq** [comment]
the code overall is good. And it is amazing to know it works!

I think the comparison part in etcdctl is hard to use. I would expect the line maybe `value(foo) == bar`.  @philips Any UX thoughts for the txn command?

**Comment by xiang90** [comment]
@yichengq We just need to make it work for now. We can focus on UX stuff later.

**Comment by xiang90** [comment]
Merging.

**Pull request merged**